### PR TITLE
Ads: Add Gutenberg/Jetpack Ads Block

### PR DESCRIPTION
**Pull request by dbspringer**
Adding a Jetpack Ads Gutenberg block per the Gutenpack Small Business Blocks initiative.

Fixes https://github.com/Automattic/wp-calypso/issues/30624

#### Changes proposed in this Pull Request:
* jetpack_register_block for wordads (`gutenblock_render`)
* Extracted/refactored `get_ad_div` into separate function to be used in `gutenblock_render`
* Added `mobile_leaderboard` to `$ad_tag_ids`, removed `lrec` (no longer supported)

#### Testing instructions

1. Compile Gutenblocks from https://github.com/Automattic/wp-calypso -- master
    1. e.g. `npm run sdk gutenberg client/gutenberg/extensions/presets/jetpack -- --output-dir="../jetpack/_inc/blocks"`
1. Add `define( 'JETPACK_BETA_BLOCKS', true );` to your `wp-config.php`
1. Enable Ads module (Jetpack Settings -> Traffic)
1. Create new post and search for "WordAds" in the Gutenberg blocks (you should see `Ad`)
    screen shot 2019-02-05 at 12 36 24 pm
1. Add some ads of various sizes and alignment, you should see them in your post 👍
screen shot 2019-02-05 at 12 39 00 pm

#### Proposed changelog entry for your changes:
* Added Ads Gutenberg block

### CODE REVIEW

**Comment by enejb** [comment]
I think we should move this code to the same place as we have the shortcode. 
Since most of the functionality of the of displaying the block would/should be the same as the shortcode.

**Comment by dbspringer** [comment]
If you think it's best that's fine, I'll move it in my next update.

File: modules/shortcodes/wordads.php [hunk 1]
```diff
@@ -53,14 +57,28 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
+			return '';
+		}
+
+		if ( $wordads->option( 'wordads_house' ) ) {
+			return $wordads->get_ad( 'inline', 'house' );
+		}
+
+		$section_id = 0 === $wordads->params->blog_id ? WORDADS_API_TEST_ID : $wordads->params->blog_id . '6';
```
Comment (enejb) [inline modules/shortcodes/wordads.php]:
```suggestion
		$section_id = ( 0 === $wordads->params->blog_id ? WORDADS_API_TEST_ID : $wordads->params->blog_id . '6' );
```

$wordads->params->blog_id . '6'   is that line intentional? Why do we need a 6. Maybe we can leave a comment explaining it.

File: modules/shortcodes/wordads.php [hunk 2]
```diff
@@ -53,14 +57,28 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
 	}
+
+	public static function gutenblock_render( $attr ) {
+		global $wordads;
+
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
+			return '';
+		}
+
+		if ( $wordads->option( 'wordads_house' ) ) {
```
Comment (enejb) [inline modules/shortcodes/wordads.php]: Would this be something that would come from the block attributes?

File: modules/wordads/wordads.php [hunk 3]
```diff
@@ -491,6 +477,40 @@ function get_ad_snippet( $section_id, $height, $width, $location = '', $css = ''
...
+	 * Returns the completed div with snipped to be inserted into the page
+	 *
+	 * @param  string  $spot top, side, inline, or belowpost
+	 * @param  string  $snippet The snippet to insert into the div
+	 * @param  array  $css_classes
+	 * @return string The supporting ad unit div
+	 *
+	 * @since 7.1
+	 */
+	function get_ad_div( $spot, $snippet, array $css_classes = null ) {
```
Comment (enejb) [inline modules/wordads/wordads.php]: Why not just default to empty array?

File: modules/wordads/wordads.php [hunk 4]
```diff
@@ -491,6 +477,40 @@ function get_ad_snippet( $section_id, $height, $width, $location = '', $css = ''
...
+
+		$css_classes[] = 'wpcnt';
+		if ( 'top' == $spot ) {
+			$css_classes[] = 'wpcnt-header';
+		}
+
+		$classes = implode( ' ', $css_classes );
+		$about  = __( 'Advertisements', 'jetpack' );
+		return <<<HTML
+		<div class="$classes">
```
Comment (enejb) [inline modules/wordads/wordads.php]: Should we be doing any `esc_* ` here?

File: modules/shortcodes/wordads.php (hunk 1 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: The various ad spots have different all used to have a unique location ids, but that's been kinda deprecated now. At this point we mostly use it to keep track of what units are getting added where so we can tell at a glance if a user or theme is doing something wonky.

File: modules/wordads/wordads.php (hunk 4 above)
Comment (dbspringer) [reply modules/wordads/wordads.php]: Added

File: modules/shortcodes/wordads.php [hunk 5]
```diff
@@ -53,14 +57,28 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		$html = '<div class="jetpack-wordad" itemscope itemtype="https://schema.org/WPAdBlock"></div>';
 		$html = $wordads->insert_inline_ad( $html );
 
 		return $html;
 	}
+
+	public static function gutenblock_render( $attr ) {
+		global $wordads;
+
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
```
Comment (enejb) [inline modules/shortcodes/wordads.php]: Lets leave a comment on this filter if it has been documented somewhere already. 
if it is a new one. Lets prefix it with `jetpack_`

**Comment by enejb** [comment]
Beside the minor change. Commenting the filter. I think this is ready to merge.

File: modules/shortcodes/wordads.php (hunk 5 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: It's been around a while:
https://github.com/Automattic/jetpack/blob/master/modules/wordads/wordads.php#L261

I'll copy the notes over

Comment (enejb) [reply modules/shortcodes/wordads.php]: Lets change the copy to instead
/** This filter is already documented in modules/wordads/wordads.php */

File: modules/shortcodes/wordads.php [hunk 6]
```diff
@@ -53,14 +57,49 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		$align = 'align' . $align;
+
+		$format = 'mrec';
+		if ( isset( $attr['format'] ) && in_array( $attr['format'], array_keys( WordAds::$ad_tag_ids ) ) ) {
+			$format = $attr['format'];
+		}
+
+		$height = WordAds::$ad_tag_ids[$format]['height'];
+		$width = WordAds::$ad_tag_ids[$format]['width'];
+		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', 'inpost', 'float:left;margin-right:5px;margin-top:0px;' );
```
Comment (enejb) [inline modules/shortcodes/wordads.php]: Do we always expect styles to be `float:left;margin-right:5px;margin-top:0px;`? 
What about when alignment is something else what about RTL?

File: modules/wordads/wordads.php [hunk 7]
```diff
@@ -491,6 +477,40 @@ function get_ad_snippet( $section_id, $height, $width, $location = '', $css = ''
...
+			$css_classes = array();
+		}
+
+		$css_classes[] = 'wpcnt';
+		if ( 'top' == $spot ) {
+			$css_classes[] = 'wpcnt-header';
+		}
+
+		$classes = esc_attr( implode( ' ', $css_classes ) );
+		$about  = __( 'Advertisements', 'jetpack' );
```
Comment (enejb) [inline modules/wordads/wordads.php]: I don't think it would hurt if we had 

```suggestion
		$about  = esc_html__( 'Advertisements', 'jetpack' );
```

File: modules/wordads/wordads.php [hunk 8]
```diff
@@ -491,6 +477,40 @@ function get_ad_snippet( $section_id, $height, $width, $location = '', $css = ''
...
+		if ( empty( $css_classes ) ) {
+			$css_classes = array();
+		}
+
+		$css_classes[] = 'wpcnt';
+		if ( 'top' == $spot ) {
+			$css_classes[] = 'wpcnt-header';
+		}
+
+		$classes = esc_attr( implode( ' ', $css_classes ) );
```
Comment (enejb) [inline modules/wordads/wordads.php]: lets also 
`$spot = esc_attr( $spot );`

File: modules/shortcodes/wordads.php (hunk 6 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: This is just something to provide a little room for the "report ad" link which gets added dynamically from the ad JS.

**Review by enejb — DISMISSED** [review]
I think this is ready to merge. Nice work.

File: modules/shortcodes/wordads.php [hunk 9]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		/** This filter is already documented in modules/wordads/wordads.php `insert_ad()` */
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
+			return '';
+		}
+
+		if ( $wordads->option( 'wordads_house' ) ) {
+			return $wordads->get_ad( 'inline', 'house' );
+		}
+
+		$section_id = 0 === $wordads->params->blog_id ? WORDADS_API_TEST_ID : $wordads->params->blog_id . '6'; // 6 is to keep track of gutenblock ads
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: I'm not sure how this will play out, could we add parens around the ternary or the concat `.` to make it clear what the order of operations is and what to expect?

File: modules/shortcodes/wordads.php [hunk 10]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		$align = 'center';
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: ✅ Should align with editor-side defaults, does:

https://github.com/Automattic/wp-calypso/pull/30565/files#diff-248676bd45f5c49457c6040408bcc1f7R38

File: modules/shortcodes/wordads.php [hunk 11]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		if ( isset( $attr['align'] ) && in_array( $attr['align'], array( 'left', 'center', 'right' ) ) ) {
+			$align = $attr['align'];
+		}
+		$align = 'align' . $align;
+
+		$format = 'mrec';
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: ✅ Should align with editor-side defaults, does:

https://github.com/Automattic/wp-calypso/pull/30565/files#diff-248676bd45f5c49457c6040408bcc1f7R42
https://github.com/Automattic/wp-calypso/pull/30565/files#diff-2d181886aeebd9d9a507342b7415cf99R6

File: modules/shortcodes/wordads.php [hunk 12]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		if ( isset( $attr['format'] ) && in_array( $attr['format'], array_keys( WordAds::$ad_tag_ids ) ) ) {
+			$format = $attr['format'];
+		}
+
+		$height = WordAds::$ad_tag_ids[$format]['height'];
+		$width = WordAds::$ad_tag_ids[$format]['width'];
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: Lint says variable array keys should be surrounded by space
```suggestion
		$width = WordAds::$ad_tag_ids[ $format ]['width'];
```

File: modules/shortcodes/wordads.php [hunk 13]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+			$align = $attr['align'];
+		}
+		$align = 'align' . $align;
+
+		$format = 'mrec';
+		if ( isset( $attr['format'] ) && in_array( $attr['format'], array_keys( WordAds::$ad_tag_ids ) ) ) {
+			$format = $attr['format'];
+		}
+
+		$height = WordAds::$ad_tag_ids[$format]['height'];
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: Lint says variable array keys should be surrounded by space
```suggestion
		$height = WordAds::$ad_tag_ids[ $format ]['height'];
```

File: modules/wordads/wordads.php [hunk 14]
```diff
@@ -491,6 +478,41 @@ function get_ad_snippet( $section_id, $height, $width, $location = '', $css = ''
 HTML;
 	}
 
+	/**
+	 * Returns the completed ad div with snipped to be inserted into the page
```
Comment (sirreal) [inline modules/wordads/wordads.php]:
```suggestion
	 * Returns the complete ad div with snippet to be inserted into the page
```

File: modules/wordads/wordads.php [hunk 15]
```diff
@@ -491,6 +478,41 @@ function get_ad_snippet( $section_id, $height, $width, $location = '', $css = ''
...
+	 * @param  string  $snippet The snippet to insert into the div
+	 * @param  array  $css_classes
+	 * @return string The supporting ad unit div
+	 *
+	 * @since 7.1
+	 */
+	function get_ad_div( $spot, $snippet, array $css_classes = array() ) {
+		if ( empty( $css_classes ) ) {
+			$css_classes = array();
+		}
```
Comment (sirreal) [inline modules/wordads/wordads.php]: Not asking for a change, but genuinely curious. What would this catch? Misuse of the method?

File: modules/shortcodes/wordads.php [hunk 16]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		$width = WordAds::$ad_tag_ids[$format]['width'];
+		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', 'float:left;margin-right:5px;margin-top:0px;' );
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: 😮 This method call makes me a bit uncomfortable 😬 

We're reproducing this?

https://github.com/Automattic/jetpack/blob/bdeb56a72746d4d00910259b1a87011e89bc7786/modules/wordads/wordads.php#L425

The reason it's duplicated is to be able to assign a different suffix? (6 not 5)?

https://github.com/Automattic/jetpack/blob/bdeb56a72746d4d00910259b1a87011e89bc7786/modules/wordads/wordads.php#L424

The partial style attribute string, especially when duplicated in two places, seems like an easy thing to screw up and get out of sync in later work. All of the other use cases for the final param are at least side by side:

https://github.com/Automattic/jetpack/blob/4f6c69e8c8d7406a68dcbb4d16f6bc17e18fc0d7/modules/wordads/wordads.php#L418-L425

Do you think we could change this so that at least we're not leaking that into this file as well?

File: modules/shortcodes/wordads.php [hunk 17]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+		return $wordads->get_ad_div( 'inline', $snippet, array( $align ) );
+	}
 }
 
 new Jetpack_WordAds_Shortcode();
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: With Gutenberg, we should start to think of shortcodes as legacy. Rather than adding the rendering here in the WordAds shortcode, how about adding this code in another spot? Maybe a new file in the WordAds module?

File: modules/shortcodes/wordads.php [hunk 18]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
+
+		$format = 'mrec';
+		if ( isset( $attr['format'] ) && in_array( $attr['format'], array_keys( WordAds::$ad_tag_ids ) ) ) {
+			$format = $attr['format'];
+		}
+
+		$height = WordAds::$ad_tag_ids[$format]['height'];
+		$width = WordAds::$ad_tag_ids[$format]['width'];
+		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', 'float:left;margin-right:5px;margin-top:0px;' );
+		return $wordads->get_ad_div( 'inline', $snippet, array( $align ) );
```
Comment (sirreal) [inline modules/shortcodes/wordads.php]: In addition to alignment, the Gutenberg editor adds a class and allows the post author to add additional classes.

If we won't use those in the output (which I suspect is intended), we should disable in the editor as well.

If you'll confirm that's intended, I'll open an issue to handle that editor-side 👍

**Review by sirreal — COMMENTED** [review]
I've reviewed the changes and left many thoughts.

File: modules/shortcodes/wordads.php [hunk 19]
```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
...
 		$html = $wordads->insert_inline_ad( $html );
 
 		return $html;
 	}
+
+	public static function gutenblock_render( $attr ) {
+		global $wordads;
+
+		/** This filter is already documented in modules/wordads/wordads.php `insert_ad()` */
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
```
[the remark this answers is not in the document]
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: This is a bit different as it's guaranteed (?) to be within the body of a post and not in some weird other place as can happen when we add the belowpost units via filters.

File: modules/shortcodes/wordads.php (hunk 9 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: I broke it up onto different lines for clarity (and because it was getting pretty long).

File: modules/shortcodes/wordads.php (hunk 12 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: Updated 👍

File: modules/shortcodes/wordads.php (hunk 13 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: Updated 👍

File: modules/wordads/wordads.php (hunk 14 above)
Comment (dbspringer) [reply modules/wordads/wordads.php]: Updated 👍

File: modules/wordads/wordads.php (hunk 15 above)
Comment (dbspringer) [reply modules/wordads/wordads.php]: Just to make sure no one passes a null or something that otherwise wouldn't get blocked by the array check. I generally just like to ensure that I have a known good value as a default for a parameter :)

File: modules/shortcodes/wordads.php (hunk 17 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: Putting the code here was a recommendation by @enejb, there seems to be some precedent in Jetpack for putting the Gutenberg blocks in the shortcode path, but I'm neutral on the subject.

File: modules/shortcodes/wordads.php (hunk 18 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: Yeah, due to requirements from our ad partners we generally want to minimize the amount of alteration our users can perform on the ad units.

File: modules/shortcodes/wordads.php (hunk 19 above)
[the remark this answers is not in the document]
Comment (sirreal) [reply modules/shortcodes/wordads.php]: > it's guaranteed (?) to be within the body of a post and not in some weird

It is at the moment, but it isn't in the long term. Gutenberg will soon be a page builder 🙂

Is that something we can take into account now so we don't need to come back and fix it later?

File: modules/shortcodes/wordads.php (hunk 16 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: This extra bit of css has to do with the special nature of the fixed locations the other ad units get inserted into (header/belowpost/sidebar/etc). Sometimes they require a little bit of extra css to ensure they slip in nicely above/below/beside other ad units and whatnot as well as have enough space for the "Report This Ad" link. Their locations/format are fixed and unlikely to change, but I still wanted a function that was generic enough to get reused in places.

Basically there's a lot of extra "stuff" going on in the original function that's not really needed for what we're doing in the block, so I wanted to streamline/simplify it. It's probable in the future that we'll remove the extra CSS bit entirely as we update our external system providing the ads to be smarter about inserting themselves.

File: modules/shortcodes/wordads.php (hunk 19 above)
[the remark this answers is not in the document]
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: I guess the point I'm making is that it's getting explicitly added to a specific spot as opposed to implicitly added via filter. Body or elsewhere, the block likely won't show up in unintended location.

File: modules/shortcodes/wordads.php (hunk 16 above)
Comment (sirreal) [reply modules/shortcodes/wordads.php]: What about some middle ground like moving the shared attribute style string into an appropriately named class constant and actually sharing it?

```suggestion
		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', WordAds::DAT_UGLY_CSS_WE_GOTTA_SHARE );
```

File: modules/shortcodes/wordads.php (hunk 17 above)
Comment (sirreal) [reply modules/shortcodes/wordads.php]: I'd say it makes more sense to be nearer to the module providing its functionality. That's what simple payments does:

https://github.com/Automattic/jetpack/blob/7b7d869dbaa6131e8782431c0f78638cc990ea98/modules/simple-payments/simple-payments.php#L43

If you think this is a more appropriate place you can leave it here.

File: modules/shortcodes/wordads.php (hunk 18 above)
Comment (sirreal) [reply modules/shortcodes/wordads.php]: https://github.com/Automattic/wp-calypso/pull/30668

File: modules/shortcodes/wordads.php (hunk 16 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: Sure thing, added `public static $SOLO_UNIT_CSS = 'float:left;margin-right:5px;margin-top:0px;';`

File: modules/shortcodes/wordads.php (hunk 17 above)
Comment (dbspringer) [reply modules/shortcodes/wordads.php]: @enejb and I seem to think that it's fine to leave here for now and until the shortcode goes away. At any rate, the refactoring seems best to go into a separate PR if needed.

Comment (sirreal) [reply modules/shortcodes/wordads.php]: That's fine 🙂

**Comment by dbspringer** [comment]
@enejb @sirreal I _thiiiink_ this is pretty well merge-ready, y'all want to give it a final review?

**Review by sirreal — DISMISSED** [review]
This looks good, thanks!

File: modules/shortcodes/wordads.php [hunk 20]
```diff
@@ -21,6 +21,10 @@ function action_init() {
 			return null;
 		}
 		add_shortcode( 'wordads', array( $this, 'wordads_shortcode' ) );
+
+		jetpack_register_block( 'wordads', array(
```
Comment (jeherve) [inline modules/shortcodes/wordads.php]: We currently use a different function (`jetpack_register_block_type`), but it should be okay to use for now as we should revert to using `jetpack_register_block` soon, as per #11328. We will, however, need to switch to using the prefix for the block name: `jetpack/wordads`

File: modules/wordads/wordads.php [hunk 21]
```diff
@@ -23,28 +23,30 @@ class WordAds {
 	 * @var array
 	 */
 	public static $ad_tag_ids = array(
-		'mrec'           => array(
+		'mrec' => array(
```
Comment (jeherve) [inline modules/wordads/wordads.php]: It seems you've lost the extra spaces that maintain the same indenting everywhere here.

File: modules/wordads/wordads.php [hunk 22]
```diff
@@ -23,28 +23,30 @@ class WordAds {
...
 			'tag'    => '300x250_mediumrectangle',
 			'height' => '250',
 			'width'  => '300',
 		),
-		'lrec'           => array(
```
Comment (jeherve) [inline modules/wordads/wordads.php]: What's the reasoning behind getting rid of that tag ID? It is not in use anymore?

File: modules/wordads/wordads.php [hunk 23]
```diff
@@ -466,11 +457,9 @@ function get_ad_snippet( $section_id, $height, $width, $location = '', $css = ''
 			'height'   => $height,
 		);
 		$ad_number   = count( $this->ads );
-		// Max 6 ads per page.
```
Comment (jeherve) [inline modules/wordads/wordads.php]: Will it be okay from now on to have many ads on a given page? There should not be any more limits?

**Review by jeherve — CHANGES_REQUESTED** [review]
This seems to work well. I only have a few questions below.

File: modules/shortcodes/wordads.php (hunk 20 above)
Comment (enejb) [reply modules/shortcodes/wordads.php]: Done

File: modules/wordads/wordads.php (hunk 22 above)
Comment (sirreal) [reply modules/wordads/wordads.php]: I believe it is no longer supported.

Description:

> * […] removed `lrec` (no longer supported)

See also: pafL3P-5e-p2 #comment-324

File: modules/wordads/wordads.php (hunk 23 above)
Comment (sirreal) [reply modules/wordads/wordads.php]: I don't believe there will be: pafL3P-5e-p2 #comment-437

cc: @rclations for confirmation

File: modules/wordads/wordads.php (hunk 21 above)
Comment (dbspringer) [reply modules/wordads/wordads.php]: The formatting was starting to look a little weird with placing the mid-level arrays on the same spacing. e.g.
```
	public static $ad_tag_ids = array(
		'mrec'               => array(
			'tag'    => '300x250_mediumrectangle',
			'height' => '250',
			'width'  => '300',
		),
		'leaderboard'        => array(
			'tag'    => '728x90_leaderboard',
			'height' => '90',
			'width'  => '728',
		),
		'mobile_leaderboard' => array(
			'tag'    => '320x50_mobileleaderboard',
			'height' => '50',
			'width'  => '320',
		),
		'wideskyscraper'     => array(
			'tag'    => '160x600_wideskyscraper',
			'height' => '600',
			'width'  => '160',
		),
	);
```
But I can put it back for consistency's sake.

File: modules/wordads/wordads.php (hunk 23 above)
Comment (dbspringer) [reply modules/wordads/wordads.php]: That's correct, the number of units is no longer limited in our back-end.

**Review by jeherve — DISMISSED** [review]
This should be good to go 👍 Merge when ready.

**Comment by dbspringer** [comment]
D'oh, committed the format change before I saw you approved the review.

**Review by jeherve — APPROVED** [review]
🚢 it!
